[1/1] Add docs for EAS Insights (#23294)
expo/expo: docs/pages/eas-insights/introduction.mdx

---
title: EAS Insights
sidebar_title: Introduction
hideTOC: true
description: An introduction to EAS Insights which is a preview service for projects using the expo-insights library.
---

**EAS Insights** is a service that will offer a view into a project's performance, usage, and reach. We are currently offering a preview of Insights that is available to all developers, and we will continue to roll out new features and functionality based on user feedback and suggestions.

EAS Insights makes it easy to see the state of your app, providing information about usage across platforms, app store versions, and timeframes.

## Integration with EAS Update

If you're already using [EAS Update](/eas-update/introduction.mdx), we provide certain high-level usage insights without any additional client-side changes. This is possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well as usage broken down by platform.

## Using the `expo-insights` module

> Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version.  Currently, the module is limited to sending client events only pertaining to cold starts of the app, but in the future we will expand `expo-insights` to offer new types of events and payloads, to support more advanced functionality.

Reviewer: ```suggestion
Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version. Currently, the module is limited to sending client events only pertaining to cold starts of the app, but in the future we will expand `expo-insights` to offer new types of events and payloads, to support more advanced functionality.
```

Small nit: extra whitespace.

---
title: EAS Insights
sidebar_title: Introduction
hideTOC: true
description: An introduction to EAS Insights which is a preview service for projects using the expo-insights library.
---

**EAS Insights** is a service that offers a view into a project's performance,
usage, and reach. We are currently offering a preview of Insights that is
available to all developers, and we will continue to roll out new features and
functionality based on user feedback and suggestions.

EAS Insights makes it easy to see the state of your app, providing information
about usage across platforms, app store versions, and timeframes.

### Integration with EAS Update

For those already using [EAS Update](/eas-update/introduction.mdx), we're able to provide certain
high-level usage insights without any additional client-side changes. This is
possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well
as usage broken down by platform.

### Using the `expo-insights` module

> Developers can add the `expo-insights` module to their project and gain more

Reviewer: ```suggestion
Developers can add the `expo-insights` library to their projects and gain more
```

> precise usage metrics (than what can be provided by just aggregating update requests), as well as additional breakdowns by app store version.

Reviewer: ```suggestion
precise usage metrics (than those provided by aggregating update requests) and additional breakdowns by app store version.
```

Currently, the module is limited to sending client events only
pertaining to cold starts of the app, but in the future we will expand
`expo-insights` to offer new types of events and payloads, to support more

> advanced functionality.

Reviewer: ```suggestion
Currently, the module is limited to sending client events only
of cold starts of the app. However, in the future, we will expand
`expo-insights` to offer new types of events and payloads to support more
advanced functionality.
```

Author: Not going to change this one, IMO the original language is preferable

---
title: EAS Insights
sidebar_title: Introduction
hideTOC: true
description: An introduction to EAS Insights which is a preview service for projects using the expo-insights library.
---

> **EAS Insights** is a service that offers a view into a project's performance,

Reviewer 2: no need limit line length, let the editor do the wrapping for you

> **EAS Insights** is a service that offers a view into a project's performance,

Reviewer 2: ```suggestion
**EAS Insights** is a service that will a view into a project's performance,
```

usage, and reach. We are currently offering a preview of Insights that is
available to all developers, and we will continue to roll out new features and
functionality based on user feedback and suggestions.

EAS Insights makes it easy to see the state of your app, providing information
about usage across platforms, app store versions, and timeframes.

### Integration with EAS Update

> For those using `expo-updates` with EAS Update, we're able to provide certain

Reviewer 2: ```suggestion
For those already using [EAS Update](/eas-update/introduction.mdx), we're able to provide certain
```

high-level usage insights without any additional client-side changes. This is

> possible due to the nature of EAS Update + expo-updates, where clients check for
> available updates and fetch them if possible. Data about these client requests
> is aggregated into a limited Insights view that shows usage over time, as well
> as usage broken down by platform.

Reviewer 2: ```suggestion
possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well
as usage broken down by platform.
```

Reviewer 2: (you can see here one of the downsides of manually wrapping docs)

> ### The New `expo-insights` Module

Reviewer 2: ```suggestion
### Using the `expo-insights` module
```

> To get the most of EAS Insights functionality, we offer the `expo-insights`
> client module. Developers can add this module to their project and gain more
> precise usage metrics, as well as additional breakdowns by app store version.
> Right now the `expo-insights` module is limited to sending client events only
> pertaining to cold starts of the app, but in the future we will expand
> `expo-insights` to offer new types of events and payloads, to support more
> advanced Insights functionality.

Reviewer 2: ```suggestion
Developers can add the `expo-insights` module to their project and gain more
precise usage metrics (than what can be provided by just aggregating update requests), as well as additional breakdowns by app store version.
Currently, the module is limited to sending client events only
pertaining to cold starts of the app, but in the future we will expand
`expo-insights` to offer new types of events and payloads, to support more
advanced functionality.
```
